tests/ArrayFunctionCallTest.php

```diff
@@ -2163,7 +2163,7 @@ function($val) { return (int)$val; },
                     );
                 ',
                 'assertions' => [
-                    '$line===' => 'array{0: int}<array-key, int>',
+                    '$line===' => 'array{0: 0}<array-key, 0>',
```

Comment:
This seems wrong. What if `$line` was `["123", "456"]`?

Reply:
Then the inferred type will be correctly set to `array{0: 123, 1: 456}`. But the given array is `["a", "b"]` which, if you typecast will be 0 for both values.

Reply:
But shouldn't `$line` be `array{0: string}<string>` when `array_map` is called? At the point the callback is evaluated it should only know that `$val` is a `string` shouldn't it?